botocore/crt/auth.py

```diff
@@ -81,7 +81,7 @@ def add_auth(self, request):
         else:
             explicit_payload = UNSIGNED_PAYLOAD
 
-        if self._should_add_content_sha256_header(explicit_payload):
+        if self._should_add_content_sha256_header(existing_sha256, explicit_payload):
```

Comment:
We should always add new positional arguments at the end of the function signature. I would also recommend making this an optional parameter that defaults to `None`.

Reply:
It looks like we also missed an update somewhere? I think this will be resolved if you follow the advice above, but we should figure out where we're missing.
````
TypeError: _should_add_content_sha256_header() takes 2 positional arguments but 3 were given
```

Reply:
I saw the error, but I don't understand why... Looks like we are not declaring `_should_add_content_sha256_header` anywhere else, and I updated the param for `_should_add_content_sha256_header` in `CrtSigV4QueryAuth`. Interesting.